db/migrate/20230830194257_add_public_email_column.rb

```diff
@@ -0,0 +1,10 @@
+class AddPublicEmailColumn < ActiveRecord::Migration[7.0]
+  def up
+    add_column :users, :public_email, :boolean, default: false
+    User.all.update_all('public_email = NOT hide_email')
```

Comment:
please do this in a maintenance_task, rather than a migration that takes a lock

Reply:
ok

Reply:
done